src/Parser.js

```diff
@@ -137,31 +142,25 @@ Parser.prototype.parseExpression = function(breakOnInfix, breakOnToken) {
     // we reached the end, a }, or a \right)
     while (true) {
         var lex = this.nextToken;
-        var pos = this.pos;
         if (endOfExpression.indexOf(lex.text) !== -1) {
             break;
         }
         if (breakOnToken && lex.text === breakOnToken) {
```

Comment:
But we break if the next *token* has the given text. True, currently we only use this for `]` which is a one-char token. But we couldn't break on `\` but we could break on `\\`, so token is more fitting. We could use `breakOnTokenText` to distinguish this from the token class, if you prefer.